drop-transfer/src/ws/server/v4.rs

```diff
@@ -527,8 +532,34 @@ impl handler::Downloader for Downloader {
                 .join(super::temp_file_name(task.xfer.id(), task.file.id())),
         );
 
+        let tmp_size = fs::File::open(&tmp_location.0)
+            .and_then(|file| file.metadata())
+            .map(|metadata| metadata.len())
+            .ok();
+
+        let will_emit_checksum_events = emit_checksum_events && tmp_size.is_some();
+
+        let cb = if will_emit_checksum_events {
+            let size = tmp_size.unwrap_or(0);
+            events.checksum_start(size).await;
+
+            Some({
+                let ev = events.clone();
+                move |progress: u64| {
+                    let ev = ev.clone();
+                    async move {
+                        ev.checksum_progress(progress).await;
+                    }
+                }
```

Comment:
These clones are not needed IMHO

Reply:
There must be the clones or else the closure is `FnOnce` as it moves the existing `ev` into it's scope

Reply:
solved